File: go/raindrops/raindrops.go
package raindrops

import (
	"fmt"
	"math/big"
	"strings"
)

// ! This is a somewhat optimized version and not meant to be idomatic

func Convert(number int) string {

	// early return any obvious values
	switch number {
	case 3:
		return "Pling"
	case 5:
		return "Plang"
	case 7:
		return "Plong"
	}

	if big.NewInt(int64(number)).ProbablyPrime(0) {
		return fmt.Sprint(number)
	}

	sounds := [3]string{"Pling", "Plang", "Plong"}
	var rDrops strings.Builder

	// grow out builder to to avoid allocation
	rDrops.Grow(15)

	for idx, fac := range [3]int{3, 5, 7} {
		if number%fac == 0 {
			fmt.Fprint(&rDrops, sounds[idx])
		}
	}

	switch res := rDrops.String(); res {
	case "":
		return fmt.Sprint(number)
	default:
		return res
	}
}

File: erlang/raindrops/src/raindrops.erl
-module(raindrops).

-export([convert/1]).

convert(Number) ->
    case {Number rem 3, Number rem 5, Number rem 7} of
        {0, 0, 0} -> "PlingPlangPlong";
        {0, 0, _} -> "PlingPlang";
        {0, _, 0} -> "PlingPlong";
        {_, 0, 0} -> "PlangPlong";
        {0, _, _} -> "Pling";
        {_, 0, _} -> "Plang";
        {_, _, 0} -> "Plong";
        _ -> integer_to_list(Number)
    end.
